hudi-spark-datasource/hudi-spark/pom.xml

```diff
@@ -267,6 +252,14 @@
       <groupId>org.apache.logging.log4j</groupId>
       <artifactId>log4j-1.2-api</artifactId>
     </dependency>
+    <dependency>
```

Comment:
We should

1. Designate all these deps as "provided" in the top-level pom
2. Add them to "hudi-common" (therefore avoiding need to replicate them across the board)
3. Override them as compile dependency in a few places (where they are not brought in by the query-engine)

Reply:
Is there a way to know which dependencies will be provided by the runtime env for each environment? My strategy right now has just been to include as compile if the tests were failing at run time due to logging.

Reply:
As discussed on Slack: all terminal modules have to have it as compile deps (cli, integ-tests, etc), the library ones have to be provided (hudi-spark-bundle, hudi-flink-bundle, etc).

Using tests is a good signal but we need to keep in mind that it's not gonna be covering everything.